drivers/gpio/gpio_sam0.c

```diff
@@ -8,13 +8,37 @@
 #include <device.h>
 #include <gpio.h>
 #include <soc.h>
+#include <interrupt_controller/sam0_eic.h>
+
+#include "gpio_utils.h"
 
 struct gpio_sam0_config {
 	PortGroup *regs;
+#ifdef CONFIG_SAM0_EIC
+	u8_t id;
+#endif
+};
+
+struct gpio_sam0_data {
+#ifdef CONFIG_SAM0_EIC
+	sys_slist_t callbacks;
+#endif
 };
 
 #define DEV_CFG(dev) \
 	((const struct gpio_sam0_config *const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct gpio_sam0_data *const)(dev)->driver_data)
+
+#ifdef CONFIG_SAM0_EIC
+static void gpio_sam0_isr(u32_t pins, void *arg)
+{
+	struct device *const dev = (struct device *) arg;
+	struct gpio_sam0_data *const data = DEV_DATA(dev);
+
+	_gpio_fire_callbacks(&data->callbacks, dev, pins);
```

Comment:
this got renamed to `gpio_fire_callbacks`

Reply:
Rebased and fixed.